Code/Framework/AzFramework/AzFramework/Network/IToolingConnection.h

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AzCore/EBus/Event.h>
+#include <AzCore/Memory/OSAllocator.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/std/containers/deque.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/smart_ptr/intrusive_ptr.h>
+#include <AzCore/std/string/string.h>
+
+namespace AzFramework
+{
+    class ToolingMessage
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(ToolingMessage, AZ::OSAllocator, 0);
+        AZ_RTTI(ToolingMessage, "{8512328C-949D-4F0C-B48D-77C26C207443}");
+
+        ToolingMessage()
```

Comment:
Consider moving all your implementational details to an .inl.. especially in an interface header like this. You're obfuscating one of the most powerful and coupled sources of documentation (the actual interface declaration itself that *must* always be kept in-sync with the code otherwise nothing would compile) by having all the implementation mixed in like this.

Reply:
That'll come in a pass once I feel the interface is relatively final.

Reply:
Should this be in a namespace of `AzFramework::Network`?

ToolMessage or ToolConnections does not really have any indication that is related to networking.
It seems more so that it is related to Editor tools.

Reply:
I've handled the inl portion. I'm still mulling over naming improvements however.